src/main/java/daomephsta/unpick/impl/AbstractInsnNodes.java

```diff
@@ -51,6 +51,16 @@ public static Object getLiteralValue(AbstractInsnNode insn)
 	
 	public static boolean isLiteral(AbstractInsnNode insn, Object literal)
 	{
-		return getLiteralValue(insn).equals(literal);
+		Object literalValue = getLiteralValue(insn);
+		// Chars are stored as ints, so conversion is necessary
+		if (literal instanceof Character && literalValue instanceof Integer)
+		{
+			Character charLiteral = (char) (int) literalValue;
+			return literal.equals(charLiteral);
+		}
+		// Compare numbers by long value, to support widening comparisons
+		if (literalValue instanceof Number && literal instanceof Number)
+		    return ((Number) literalValue).longValue() == ((Number) literal).longValue();
```

Comment:
I wonder if this works well for float/double comparisons

Reply:
The tests all pass, so better than you might think apparently. 
I will definitely correct this oversight though.

Reply:
I mean if we have a 0.8 and a 0.6, do their long value both become 0 as they round down?

Reply:
Tests passing doesn't mean it works, it could also mean the tests don't cover this test case.

Reply:
I know. I was mentioning it offhand. As I said, regardless of test status, it's an oversight that needs to be fixed.